hadoop-hdfs-project/hadoop-hdfs-rbf/src/main/java/org/apache/hadoop/hdfs/server/federation/router/NamenodeHeartbeatService.java

```diff
@@ -482,17 +508,36 @@ private void updateHAStatusParameters(NamenodeStatusReport report) {
     }
   }
 
+  /**
+   * Evaluates whether the JMX report should be refreshed by
+   * calling the Namenode, based on the following conditions:
+   * 1. JMX Updates must be enabled.
+   * 2. The last attempt to update JMX occurred before the
+   *    configured interval (if any).
+   */
+  private boolean shouldUpdateJmx() {
+    return updateJmxEnabled
+        && (updateJmxIntervalMs < 0
+        || Time.monotonicNow() - lastJmxUpdateAttempt > updateJmxIntervalMs);
```

Comment:
can we only rely on one config rather than both? here we can get rid of `updateJmxEnabled` and only rely on `updateJmxIntervalMs`?

something like:

if updateJmxIntervalMs < 0   => return false (disabled)
if updateJmxIntervalMs == 0 => return true (enabled)
if updateJmxIntervalMs > 0   => return `Time.monotonicNow() - lastJmxUpdateAttempt > updateJmxIntervalMs`

Reply:
I agree that we can remove updateJmxEnabled without it being too confusing. Thanks